### ISSUE TITLE
Torch object expected error on getParameters

### ISSUE THREAD

**Issue by szagoruyko** [fmassa/optimize-net#7]
Wanted to give it a try on imagenet and got this, errors out on getParameters.
```
luajit: Module.lua:218: Torch object, table, thread, cdata or function expected
stack traceback:
	[C]: in function 'pointer'
	Module.lua:218: in function 'flatten'
	Module.lua:292: in function 'getParameters'
	train.lua:148: in main chunk
```

**Comment by fmassa** [comment]
Did you set the training mode for the optimizer ?

**Comment by fmassa** [comment]
If you want to use it in evaluation mode and call getParameters, you should disable the gradParameter zeroing, by using the option `removeGradParameters=true`.

**Comment by szagoruyko** [comment]
yes works this way, missed this option in readme. thanks

**Comment by fmassa** [comment]
Thanks for opening the issue. I'll improve the README with some more detailed information about the available options in optimizeMemory

**Comment by fmassa** [comment]
Just a heads up on this, https://github.com/fmassa/optimize-net/commit/1f69144c3156be8d4dc971987e1ad214eb4b5c9b adds a note in the README about possible optimization options, https://github.com/fmassa/optimize-net/commit/7e698ef3f5a3ed1e0967481aa2e2f497b7eac395 disables `getParameters` in inference mode, and https://github.com/fmassa/optimize-net/commit/ea0b101365e077d9f337951210d4ec1613f6fd0b avoids double memory optimization. Those changes should hopefully make using `optnet` less error prone.
Thanks again @szagoruyko for opening this issue !

**Comment by meijie0401** [comment]
@fmassa Dear friend, I met the same problem. How to using the option removeGradParams=false in code?

**Comment by meijie0401** [comment]
@szagoruyko Dear friend, I met the same problem. What change did you make in your code?

**Comment by fmassa** [comment]
I suppose all you should do is to pass `removeGradParams=false` in the code?
But I've not used/maintained this library for a long time, so things are not fresh in my mind.
I'd recommend using PyTorch now, because it contains better memory savings than what this library provides.
